2017-02-07  Amit Ballav Mondal  <amit@lnxsrv07.seas.ucla.edu>

	* timespec.h (timespec_cmp) - avoid integer overflow with direct
	comparisons instead of stored values.

--- lib/timespec.h	2017-02-07 15:22:24.817648000 -0800
+++ lib/timespec.h	2017-02-07 16:13:36.033061000 -0800
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec) return 1;
+  else if (a.tv_sec < b.tv_sec) return -1;
+  else return a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
